quilt/tools/store.py

```diff
@@ -161,6 +161,34 @@ def temporary_object_path(self, name):
         """
         return os.path.join(self._path, self.TMP_OBJ_DIR, name)
 
+def parse_package_extended(name):
+    hash = version = tag = None
+    try:
+        if ':' in name:
+            name, versioninfo = name.split(':', 1)
+            if ':' in versioninfo:
+                info = versioninfo.split(':', 1)
+                if len(info) == 2:
+                    if info[0][0] == 'v':
```

Comment:
This looks a bit overly permissive since you're only checking the first letter. It's probably worth explicitly checking for 't' or 'tag' and 'v' or 'version'.

Reply:
NOCHG: aneesh requested this.